packages/core/src/electron-main/electron-main-application.ts

```diff
@@ -287,18 +300,87 @@ export class ElectronMainApplication {
         return this.didUseNativeWindowFrameOnStart.get(webContents.id) ? 'native' : 'custom';
     }
 
+    /**
+     * Shows the splash screen, if it was configured. Otherwise does nothing.
+     */
+    protected showSplashScreen(): void {
+        if (this.isShowSplashScreen()) {
+            console.log('Showing splash screen');
+            const splashScreenOptions = this.getSplashScreenOptions();
+            if (!splashScreenOptions) {
+                // sanity check, should always exist here
+                return;
+            }
+            const content = splashScreenOptions.content;
+            console.debug('SplashScreen options', splashScreenOptions);
+            app.whenReady().then(() => {
+                const splashScreenBounds = this.determineSplashScreenBounds();
+                const splashScreenWindow = new BrowserWindow({
+                    ...splashScreenBounds,
+                    frame: false,
+                    alwaysOnTop: true,
+                });
+                splashScreenWindow.show();
+                splashScreenWindow.loadFile(path.resolve(this.globals.THEIA_APP_PROJECT_PATH, content).toString());
+                this.splashScreenState = {
+                    splashScreenWindow,
+                    minTime: new Promise(resolve => setTimeout(() => resolve(), splashScreenOptions.minDuration ?? 0)),
+                    maxTime: new Promise(resolve => setTimeout(() => resolve(), splashScreenOptions.maxDuration ?? 60000)),
+                };
+            });
+        }
+    }
+
+    protected determineSplashScreenBounds(): { x: number, y: number, width: number, height: number } {
+        const splashScreenOptions = this.getSplashScreenOptions();
+        const width = splashScreenOptions?.width ?? 640;
```

Comment:
640x480? Is VGA still a thing? But seriously: I was wondering why we always have to give a size for the window. Does the browser window not adjust automatically to the size of the loaded file?

Reply:
As we don't know what's in the content we can't really center it on screen without using some predetermined width/height. I'm open for other defaults or if you know some better way of choosing on which screen the splash shall appear without using x, y, width and height.

Reply:
Does the window not auto-resize to fit the content? Maybe we can load the file and then query the window dimensions and position the window accordingly and only then do the `show`? https://www.electronjs.org/docs/latest/api/browser-window#showing-the-window-gracefully. 640x480 seems so arbitrary.

Reply:
Generally speaking in the web it works the other way around, the content fits to the browser size.

In theory we could add some sort of analyzing to try to determine the content width. I guess we could use `window.webContents.executeJavascript(javascriptAsString)` and then try to implement some heuristic size guessing, falling back to our default size in the cases where we can't. I guess the only sizes where it's 100% sure are the ones where the content is really just a png or someone who set a max width (and max height) in their styles.

The downside of this approach is that it obstructs what the splash screen is for: To be shown as early as possible to indicate to the user that the app is loading. Instead of showing it as early as possible, we first need to render it non-visible, then execute the analysis Javascript, move the window into the correct position, resize and only then show it.

Alternative suggestion: Instead of hard coding the splash screen size default we could calculate the default based on the screen size, similar to the Theia window size default. Would you prefer that solution?